core/domain/learner_playlist_services_test.py

```diff
@@ -257,6 +291,45 @@ def test_mark_collection_to_be_played_later(self):
             self._get_all_learner_playlist_collection_ids(
                 self.user_id), collection_ids)
 
+        # Empty the learner playlist again.
+        for collection_id in collection_ids:
+            learner_playlist_services.remove_collection_from_learner_playlist(
+                self.user_id, collection_id)
+        self.assertEqual(
+            self._get_all_learner_playlist_collection_ids(
+                self.user_id), [])
+
+        # Subscribe to collection.
+        subscription_services.subscribe_to_collection(
+            self.user_id, 'SUBSCRIBED_ID')
+
+        # Now if we try to add the same collection, it shouldn't be added.
+        learner_progress_services.add_collection_to_learner_playlist(
+            self.user_id, 'SUBSCRIBED_ID')
+        self.assertEqual(
+            self._get_all_learner_playlist_collection_ids(
+                self.user_id), [])
+
+        # Add MAX_LEARNER_PLAYLIST_ACTIVITY_COUNT collections at given
+        # positions.
+        for i in range(0, MAX_LEARNER_PLAYLIST_ACTIVITY_COUNT):
+            learner_progress_services.add_collection_to_learner_playlist(
+                self.user_id, collection_ids[i], position_to_be_inserted=i)
+        self.assertEqual(
+            self._get_all_learner_playlist_collection_ids(
+                self.user_id), collection_ids)
+
+        # Now if we try to add another exploration at any given position, it
+        # shouldn't be added as the list length would exceed
+        # MAX_LEARNER_PLAYLIST_ACTIVITY_COUNT.
+        learner_playlist_services.mark_collection_to_be_played_later(
```

Comment:
Similar changes might be needed here, as mentioned above.

Reply:
Same as above.